src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,429 @@
+import { Gesture } from './gesture';
+import { GestureBuilder, BuiltGesture } from './gestureBuilder';
+import { Directions } from '../../Directions';
+import {
+  GestureEventPayload,
+  HandlerStateChangeEventPayload,
+} from '../gestureHandlerCommon';
+import { getNextHandlerTag } from '../handlersRegistry';
+import { PanGestureHandlerEventPayload } from '../PanGestureHandler';
+import { ForceTouchGestureHandlerEventPayload } from '../ForceTouchGestureHandler';
+import { PinchGestureHandlerEventPayload } from '../PinchGestureHandler';
+import { RotationGestureHandlerEventPayload } from '../RotationGestureHandler';
+
+type HitSlopType =
+  | number
+  | Partial<
+      Record<
+        'left' | 'right' | 'top' | 'bottom' | 'vertical' | 'horizontal',
+        number
+      >
+    >
+  | Record<'width' | 'left', number>
+  | Record<'width' | 'right', number>
+  | Record<'height' | 'top', number>
+  | Record<'height' | 'bottom', number>;
+
+type CommonGestureConfig = {
+  ref: React.RefObject<SimpleGesture>;
+  enabled: boolean;
+  minPointers: number;
+  shouldCancelWhenOutside: boolean;
+  hitSlop: HitSlopType;
+  requireToFail: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+  simultaneousWith: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+};
+
+type PanGestureHandlerEvent = GestureEventPayload &
+  PanGestureHandlerEventPayload;
+
+type PinchGestureHandlerEvent = GestureEventPayload &
+  PinchGestureHandlerEventPayload;
+
+type RotationGestureHandlerEvent = GestureEventPayload &
+  RotationGestureHandlerEventPayload;
+
+type ForceTouchGestureHandlerEvent = GestureEventPayload &
+  ForceTouchGestureHandlerEventPayload;
+
+type HandlerCallbacks = {
+  handlerTag: number;
+  onBegan?: (event: HandlerStateChangeEventPayload) => void;
+  onStart?: (event: HandlerStateChangeEventPayload) => void;
+  onEnd?: (event: HandlerStateChangeEventPayload, success: boolean) => void;
+  onUpdate?:
+    | ((event: PanGestureHandlerEvent) => void)
+    | ((event: PinchGestureHandlerEvent) => void)
+    | ((event: RotationGestureHandlerEvent) => void)
+    | ((event: ForceTouchGestureHandlerEvent) => void);
```

Comment:
Those are plain events or events wrapped with `{nativeEvent: eventData}`?

Reply:
I'd like to unify types from here and gestureHandlerCommon to have them in one place. Also, please name this type since it's not clear if only handlers with continuous gestures should belong here or somewhere else.

Reply:
Those are plain events. I refactored it to use generics in 712e64fc3e4ab072ac817af3700ba4918d277b12 and it uses the same payload types as old handlers. Is that ok?